src/main/java/org/corfudb/infrastructure/ManagementServer.java

```diff
@@ -0,0 +1,236 @@
+     * Wait until we receive a message with the layout endpoint to bootstrap the management server.
+     */
+    private Layout waitForBootstrap() {
+        Layout currentLayout = null;
+        while (currentLayout == null) {
+            try {
+                Thread.sleep(1000);
+                currentLayout = layoutServer.getCurrentLayout();
+            } catch (InterruptedException err) {
+                log.warn("Thread interrupted.");
+                if (isShutdown) {
+                    return null;
+                }
+            }
+        }
+        log.info("Node bootstrapped. Layout received by management server.");
+        return currentLayout;
+    }
+
+    /**
+     * Returns a connected instance of the CorfuRuntime.
+     *
+     * @return A connected instance of runtime or null if isShutdown is true.
+     */
+    private synchronized CorfuRuntime getCorfuRuntime() {
+
+        if (corfuRuntime == null) {
+
+            // TODO: Check for persisted layout if saved by the management server.
+            // If yes, connect runtime to that layout as it is the latest snapshot
+            // of the layout servers we have.
+            // If no persisted layout present, wait for bootstrap. (Rely on local layout server)
+
+            Layout bootstrapLayout = waitForBootstrap();
+            if (bootstrapLayout == null || isShutdown) {
+                return null;
+            }
+
+            corfuRuntime = new CorfuRuntime();
+            bootstrapLayout.getLayoutServers().forEach(ls -> corfuRuntime.addLayoutServer(ls));
+            corfuRuntime.connect();
+        }
+        return corfuRuntime;
+    }
+
+
+    /**
+     * Gets the address of this endpoint.
+     *
+     * @return localEndpoint address
+     */
+    private String getLocalEndpoint() {
+        return this.opts.get("--address") + ":" + this.opts.get("<port>");
+    }
+
+
+    /**
+     * This contains the complete failure detection and handling mechanism.
+     * <p>
+     * It first checks whether the current node is bootstrapped.
+     * If not, it continues checking in intervals of 1 second.
+     * If yes, it sets up the corfuRuntime and continues execution
+     * of the policy.
+     * <p>
+     * It executes the policy which detects and reports failures.
+     * Once detected, it triggers the trigger handler which takes care of
+     * dispatching the appropriate handler.
+     * <p>
+     * Currently executing the periodicPollPolicy.
+     * It executes the the polling at an interval of every 1 second.
+     * After every poll it checks for any failures detected.
+     */
+    private void taskScheduler() {
+
+        Layout currentLayout;
+        LayoutView layoutView;
+        CorfuRuntime corfuRuntime;
+
+        do {
+            corfuRuntime = getCorfuRuntime();
+            if (isShutdown) return;
+        } while (corfuRuntime == null);
+
+
+        while (true) {
+            try {
+                corfuRuntime.invalidateLayout();
```

Comment:
Why are you invoking invalidateLayout on each polling interval? Unless I am missing something, this is wrong. It will unnecessarily cause an epoch-increase on each polling internal.